plugins/module_utils/network/sonic/config/interfaces/interfaces.py

```diff
@@ -192,18 +193,23 @@ def _state_overridden(self, want, have, diff):
                   to the desired configuration
         """
         commands = []
-        commands_del = self.filter_comands_to_change(want, have)
-        requests = self.get_delete_interface_requests(commands_del, have)
-        del_req_count = len(requests)
-        if commands_del and del_req_count > 0:
-            commands_del = update_states(commands_del, "deleted")
-            commands.extend(commands_del)
-
-        commands_over = diff
-        requests.extend(self.get_modify_interface_requests(commands_over, have))
-        if commands_over and len(requests) > del_req_count:
-            commands_over = update_states(commands_over, "overridden")
-            commands.extend(commands_over)
+        requests = []
+        if not diff:
+            return commands, requests
```

Comment:
This may not correct: if want is subset of have, diff=[], so the code does nothing. However in that case, want should override have for "overridden" state.

Similarly for the above "_state_replaced" function (could not add comment at it - line#76, so I add comment here). if the config for an interface in want is subset of those in have, the config for the interface in want should replace the config for same interface in have.

Reply:
corrected both functions.